engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/selectModulesScreen/SelectModulesScreen.java

```diff
@@ -407,13 +407,56 @@ public String get() {
                 }
                 filterModules();
             });
+
+            UIButton resetAdvancedFilters = find("resetFilters", UIButton.class);
+            filterModules();
+            if (resetAdvancedFilters != null) {
+
```

Comment:
It seems like we have a lot of duplicated code now for keeping checkboxes in sync with the data.
If I remember correctly, it's possible to bind a NUI control to a data source.   I recommend that we do that rather than explicitly setting both the data source and then the checkbox.  Once weve bound the checkbox to the appropriate data source, all we should have to do is set the value on the data source.

I also think that the default values on the data source should be set by a method in the data source.   Right now you have to parse code to determine what the default values are.   (Figure out what the state of the data is at the time of the if statement that toggles the value....)

Instead, we should call selectModulesConfig.resetToDefaultValues().   Then the checkboxes should pick up the changes through bindings automatically.

Reply:
Yes, I tested this and it is much cleaner this way.

For instance, the ui code for local-only can be reduced to this.  I also fixed the getter/setter to work with boolean not Boolean:
`
            UICheckbox localOnlyCheckbox = find("localOnlyCheckbox", UICheckbox.class);
            localOnlyCheckbox.bindChecked(BindHelper.bindBeanProperty("localOnlySelected", selectModulesConfig, boolean.class));
            localOnlyCheckbox.subscribe(e -> {
                filterModules();
            });
`

Reply:
Creating a binding for the other dynamic checkboxes will require a custom binding subclass (modification of BeanBinding).    

The button action for uncategorized with a boolean bean binding needs more thought.   Probably the logic in the button action would be moved inside the selectModulesConfig.setUncategorizedSelected() method.

I can apply this PR as-is (once the extra filterModule(); method call is removed) if you are not interested in changing the code to use bindings.  I can go ahead and do that in a separate PR.